app/src/main/res/layout/activity_detail.xml

```diff
@@ -44,19 +44,39 @@
                     android:paddingEnd="12dp"
                     android:paddingStart="12dp"
                     android:paddingTop="6dp"
-                    android:textAppearance="@style/TextAppearance.AppCompat.Body1" />
+                    android:textAppearance="@style/TextAppearance.AppCompat.Body1"/>
 
             </RelativeLayout>
 
             <android.support.v7.widget.Toolbar
                 android:id="@+id/toolbar"
                 android:layout_width="match_parent"
-                android:layout_height="50dp" />
+                android:layout_height="50dp"/>
 
         </android.support.design.widget.CollapsingToolbarLayout>
 
     </android.support.design.widget.AppBarLayout>
 
+    <android.support.design.widget.FloatingActionButton
```

Comment:
This buttons should be on the bottom of the xml as they may be hidden under other ui elements on older devices.